test/functional/plugin/shada_spec.lua

```diff
@@ -7,12 +9,21 @@ local read_file = helpers.read_file
 
 local mpack = require('mpack')
 
-local plugin_helpers = require('test.functional.plugin.helpers')
-local reset = plugin_helpers.reset
-
 local shada_helpers = require('test.functional.shada.helpers')
 local get_shada_rw = shada_helpers.get_shada_rw
 
+local function reset(shada_file)
+  -- TODO(justinmk): why is this needed?
+  local rtp_value = ('\'%s/runtime\''):format(
+      paths.test_source_path:gsub('\'', '\'\''))
+  clear{args_rm={'-u', '-i'},
+        args={'-u', 'NORC',
+              '-i', shada_file or 'NONE',
+              '--cmd', 'set laststatus&',
+              '--cmd', 'let &runtimepath='..rtp_value,
+        }}
+end
```

Comment:
This also works:
```diff
diff --git i/test/functional/plugin/shada_spec.lua w/test/functional/plugin/shada_spec.lua
index 778dc4e21..461f5b3b9 100644
--- i/test/functional/plugin/shada_spec.lua
+++ w/test/functional/plugin/shada_spec.lua
@@ -13,14 +13,9 @@ local shada_helpers = require('test.functional.shada.helpers')
 local get_shada_rw = shada_helpers.get_shada_rw

 local function reset(shada_file)
-  -- TODO(justinmk): why is this needed?
-  local rtp_value = ('\'%s/runtime\''):format(
-      paths.test_source_path:gsub('\'', '\'\''))
-  clear{args_rm={'-u', '-i'},
-        args={'-u', 'NORC',
+  clear{ args={'-u', 'NORC',
               '-i', shada_file or 'NONE',
               '--cmd', 'set laststatus&',
-              '--cmd', 'let &runtimepath='..rtp_value,
         }}
 end
```

You've added the `'set laststatus&'` there - why does it change the outcome of
1 test ("syntax/shada.vim works")?

```diff
--- /tmp/1expected      2019-08-06 12:56:47.920825416 +0200
+++ /tmp/1passedin      2019-08-06 12:56:54.517412573 +0200
@@ -664,15 +664,18 @@
  *[23] = {
    *[1] = {
      *[1] = {
-       *[1] = 'ShaDaEntryArrayEntryStart' }
+       *[1] = 'ShaDaEntryArray'
+        [2] = 'ShaDaEntryArrayEntryStart' }
       [2] = '  - ' }
     [2] = {
       [1] = {
-        [1] = 'ShaDaEntryArrayDescription' }
+        [1] = 'ShaDaEntryArray'
+        [2] = 'ShaDaEntryArrayDescription' }
       [2] = ':s replacement string  ' }
     [3] = {
       [1] = {
-        [1] = 'ShaDaMsgpackShaDaKeyword' }
+        [1] = 'ShaDaEntryArray'
+        [2] = 'ShaDaMsgpackShaDaKeyword' }
       [2] = 'DEBUG' } }
   [24] = {
     [1] = {
```